File: libraries/cyclestreets-view/src/main/java/net/cyclestreets/routing/ReplanRoutingTask.java
package net.cyclestreets.routing;

import android.content.Context;
import net.cyclestreets.view.R;
import net.cyclestreets.content.RouteData;
import net.cyclestreets.content.RouteDatabase;

public class ReplanRoutingTask
  extends RoutingTask<Journey>
{
	private final RouteDatabase db_;
	private final String newPlan_;

	ReplanRoutingTask(final String newPlan,
	                  final RouteDatabase db,
	                  final Context context)
	{
		super(R.string.route_loading, context);
		db_ = db;
		newPlan_ = newPlan;
	} // ReplanRouteTask

	@Override
	protected RouteData doInBackground(Journey... params)
	{
	  final Journey pr = params[0];
	  final RouteData rd = db_.route(pr.itinerary(), newPlan_);
	  if(rd != null)
		  return rd;

	  publishProgress(R.string.route_finding_new);
	  return fetchRoute(newPlan_, pr.itinerary(), 0, pr.waypoints());
	} // doInBackground
} // class ReplanRoutingTask

File: libraries/cyclestreets-view/src/main/java/net/cyclestreets/routing/ReplanRoutingTask.kt
package net.cyclestreets.routing

import android.content.Context

import net.cyclestreets.content.RouteData
import net.cyclestreets.content.RouteDatabase
import net.cyclestreets.view.R

internal class ReplanRoutingTask(private val newPlan: String,
                                 private val db: RouteDatabase,
                                 context: Context) : RoutingTask<Journey>(R.string.route_loading, context) {
    override fun doInBackground(vararg params: Journey): RouteData? {
        val pr = params[0]
        val rd = db.route(pr.itinerary(), newPlan)
        if (rd != null)
            return rd

        publishProgress(R.string.route_finding_new)
        return fetchRoute(newPlan, pr.itinerary().toLong(), 0, pr.waypoints)
    }
}
